src/java/org/apache/cassandra/db/streaming/ComponentManifest.java

```diff
@@ -52,13 +53,17 @@ public ComponentManifest(Map<Component, Long> components)
     }
 
     @VisibleForTesting
-    public static ComponentManifest create(Descriptor descriptor)
+    public static ComponentManifest create(SSTable sstable)
     {
-        LinkedHashMap<Component, Long> components = new LinkedHashMap<>(descriptor.getFormat().streamingComponents().size());
+        Set<Component> streamingComponents = sstable.getComponents();
+        LinkedHashMap<Component, Long> components = new LinkedHashMap<>(streamingComponents.size());
 
-        for (Component component : descriptor.getFormat().streamingComponents())
+        for (Component component : streamingComponents)
         {
-            File file = descriptor.fileFor(component);
+            if (component == SSTableFormat.Components.TOC)
```

Comment:
nit: I guess the spirit of the original code here was to delegate this as much as possible to the SSTable format. What if we tried to preserve that (and coupled it with the `SAI` component I suggested below) and did something like...

```
if (!sstable.descriptor.getFormat().streamingComponents().contains(component) && component.type != BigFormat.Components.Types.SAI)
```
I like the idea of using `sstable.getComponents()` to make sure we don't forget anything, but it seems like picking out the `TOC` alone is a little brittle. WDYT?

Reply:
We could also hide this in `SSTable` with a `getStreamingComponents()` and leave this logic in the manifest almost unchanged?

Reply:
First, I like the idea of having `getStreamingComponents` method on the sstable. This looks like a better place to tell which components are eligible for streaming. 

However, even with such method, we still have a "responsibility leak" from SAI here. 

```
c.type == Components.Types.SAI
```

Why should SSTable code know anything about SAI? Why should it assume all SAI components should be streamed?  IMHO this is something that should be decided by the SAI code (or whoever owns the component), not by sstable code.

SAI already controls the type of the component, so I guess the ideal-world solution would be to have this information associated directly with the type so instead of special-casing SAI vs core components, we could write the condition like this:

```
 return components.stream()
                     .filter(c -> c.type.shouldBeStreamed)
                     .collect(Collectors.toSet());
```

WDYT?

Reply:
Yeah, at the end of the day, some things are just not part of the SSTable format, so if/when that is the case, having the component decide whether it's "streamable" doesn't seem like a bad idea. (Or we could do that for all components.)